Sources/Utility/Process.swift

```diff
@@ -373,6 +373,98 @@ extension Process {
     }
 }
 
+// MARK: utility function for Process
+
+// NOTE FOR PR: follow on from bugfix PR comments https://github.com/apple/swift-package-manager/pull/921/files
+
+// roughly matches the functionality from llbuild at lib/llvm/Support/Unix/Program.inc
+//public enum ProgramSearchError: Swift.Error {
+//    case couldNotFindProgram(let program):
+//        print(error: "Unable to find \(program).")
+//    case ProgramNotExecutable(let program):
+//        print(error: "Program found at \(program) does not have permissions to be executed.")
+//}
+
+//public func findProgramByName(_ name: String, paths: [String]) -> String throws {
+// if paths is empty, utilize os.env['PATH'] and convert into a set of paths to search
+// should it return AbsolutePath instead of a String?
+// should it throw an exception is the program isn't found?
+
+    //precondition(name, "Must provide a program name")
+    // (heckj) double check how to handle default values for something like paths [] - ideally leave it as optional parameter
+
+    //var environmentPath = ProcessInfo.processInfo.environment["PATH"]
+    //(type of environmentPath is String?)
+    //let paths = oldPath.components(separatedBy: ":")
+        // if the name starts with a '/', return it explicitly if it exists and can be executed
+
+        // iterate through paths
+            // if exists && can_execute, then return path to it
+            // else throw ProgramSearchError.couldNotFindProgram
+//}
+
+/// Create a list of AbsolutePath search paths from a string, such as the PATH environment variable.
+///
+/// - Parameters:
+///   - pathString: The path string to parse.
+///   - currentWorkingDirectory: The current working directory, the relative paths will be converted to absolute paths based on this path.
+/// - Returns: List of search paths.
+public func getEnvSearchPaths(
```

Comment:
another PR for the migration of the search paths bits from UserControls -> Process coming up